lib/kubernetes-deploy/restart_task.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+require 'kubernetes-deploy/kubeclient_builder'
+
+module KubernetesDeploy
+  class RestartTask
+    include KubernetesDeploy::KubeclientBuilder
+
+    HTTP_OK_RANGE = 200..299
+
+    def initialize(context:, namespace:, logger:)
+      @context = context
+      @namespace = namespace
+      @logger = logger
+      @v1beta1_kubeclient = build_v1beta1_kubeclient(context)
+    end
+
+    def perform(deployments)
+      if deployments.empty?
+        raise ArgumentError, "#perform takes at least one deployment to restart"
+      end
+
+      fetch_deployments(deployments).each do |record|
+        response = patch_deployment_with_restart(record)
+        if HTTP_OK_RANGE.cover?(response.code)
+          @logger.info "Restarted #{record.metadata.name.inspect}"
+        else
+          @logger.fatal "Failed to restart #{record.metadata.name}. API returned non-200 response code (#{response.code})"
+          @logger.fatal "Response:"
+          @logger.fatal response.body
+        end
+      end
+    end
+
+    private
+
+    def build_v1beta1_kubeclient(context)
+      build_kubeclient(
+        api_version: "v1beta1",
+        context: context,
+        endpoint_path: "/apis/extensions/"
+      )
+    end
+
+    def patch_deployment_with_restart(record)
+      @v1beta1_kubeclient.patch_deployment(
+        record.metadata.name,
+        build_patch_payload(record),
+        @namespace
+      )
+    end
+
+    def fetch_deployments(list)
+      list.map do |name|
+        records = @v1beta1_kubeclient.get_deployments(label_selector: "name=#{name}", namespace: @namespace)
```

Comment:
I think this code isn't handling the case where the deployment name is bad / not found for whatever reason. In any case I think we should use `@v1beta1_kubeclient.get_deployment(name, @namespace)` and possibly capture/re-raise the `KubeException` we'd get on 404 (if the resulting message would suck otherwise). Although Shopify's templates label all resources with a name label matching the resource name by convention, we don't actually control the templates, so we shouldn't rely on it (it's fine in tests since we do control those).